src/python/deepgnn/graph_engine/snark/meta_merger.py

```diff
@@ -48,97 +49,81 @@ def __init__(self, fs: fsspec.AbstractFileSystem, path: str) -> None:
         self._from_file()
 
     def _from_file(self):
-        with self.fs.open(self.path, "r") as meta:
-            self.version = meta.readline().strip()
-            if self.version[0] != "v":
-                raise RuntimeError(
-                    "First line in meta file should be version, please regenerate binary data"
-                )
-
-            self.node_count = int(meta.readline())
-            self.edge_count = int(meta.readline())
-            self.node_type_num = int(meta.readline())
-            self.edge_type_num = int(meta.readline())
-            self.node_feature_count = int(meta.readline())
-            self.edge_feature_count = int(meta.readline())
-            self.partition_count = int(meta.readline())
-            self.partition_ids = []
-            self.node_weights = [
-                [float(0)] * self.node_type_num for _ in range(self.partition_count)
-            ]
-            self.edge_weights = [
-                [float(0)] * self.edge_type_num for _ in range(self.partition_count)
-            ]
-            for p in range(self.partition_count):
-                self.partition_ids.append(int(meta.readline()))  # Partition id
-                for i in range(self.node_type_num):
-                    self.node_weights[p][i] = float(meta.readline())
-                for i in range(self.edge_type_num):
-                    self.edge_weights[p][i] = float(meta.readline())
-
-            self.node_count_per_type = []
-            self.edge_count_per_type = []
-
-            for _ in range(self.node_type_num):
-                self.node_count_per_type.append(int(meta.readline()))
-            for _ in range(self.edge_type_num):
-                self.edge_count_per_type.append(int(meta.readline()))
-
-    def get_partition(self):
+        with self.fs.open(self.path, "r") as file:
+            meta = json.load(file)  # type: ignore
+
+        self.version = meta["binary_data_version"]
+        if self.version[0] != "v":
+            raise RuntimeError(
+                "Meta file should contain binary_data_version, please regenerate binary data"
+            )
+
+        self.node_count = int(meta["node_count"])
+        self.edge_count = int(meta["edge_count"])
+        self.node_type_num = int(meta["node_type_num"])
+        self.edge_type_num = int(meta["edge_type_num"])
+        self.node_feature_count = int(meta["node_feature_num"])
+        self.edge_feature_count = int(meta["edge_feature_num"])
+        self.partition_count = int(meta["n_partitions"])
+        self.partition_ids = meta["partition_ids"]
+        self.node_weights = [[float(0)] * self.node_type_num] * self.partition_count
+        self.edge_weights = [[float(0)] * self.edge_type_num] * self.partition_count
+        for j, id in enumerate(self.partition_ids):
+            for i in range(self.node_type_num):
+                self.node_weights[j][i] += float(meta[f"node_weight_{id}"][i])
```

Comment:
It looks like you keep meta structure flat. Please, unroll lists and make nested objects. Something like this:
`{"binary_version": "v2", "partitions": [{"id":0, "node_count": 123, "node_weight":...}]`. This will make merging and preprocessing way easier.

Reply:
The only values that are per partition are node and edge weights, I can make these into nested lists and leave the rest?

Reply:
Okay I think we should get the format down exactly before I change it since it'll take days to change. Right now I will make it look like this:
{"binary_data_version": "v2", "node_count": 2708, "edge_count": 10556, "node_type_num": 1, "edge_type_num": 1, "node_feature_num": 2, "edge_feature_num": 0, "partitions": {0: {"node_weight": [2708], "edge_weight": [10556.0]}}, "node_count_per_type": [2708], "edge_count_per_type": [10556]}

Reply:
Looks good.

Reply:
I've updated the C++ to include this for now.